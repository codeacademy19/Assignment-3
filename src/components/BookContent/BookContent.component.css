.mainContent{
    justify-self: center;
    display: flex;
    flex-direction: column;
    max-width: 350px;
}

.main{
    padding: 0%;
}
.backgroundImage{

    width: 350px;
    height: 300px;
    background-size: cover;
}

.title{

    display:flex;
    justify-content: space-between;
    font-size:1rem;
    padding:10px;
    text-align: center;
    height:27px;

}

