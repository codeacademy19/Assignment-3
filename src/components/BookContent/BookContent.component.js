import React, { Component } from 'react';
import './BookContent.component.css';

class BookContent extends Component {
  render() {
    const book = this.props.book;

    return (
      <div className="mainContent">
        <div className="backgroundImage" style={{ backgroundImage: 'url("./Images/cover_image.jpeg")' }} />

        <div className="title">

          <span>{book.Name.toUpperCase()}</span>

        </div>
      </div>
    );
  }
}
export default BookContent;
