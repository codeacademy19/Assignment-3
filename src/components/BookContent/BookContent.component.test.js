import React from 'react';
import renderer from 'react-test-renderer';
import BookContent from './BookContent.component';


describe('AllBooksOfAnAuthor', () => {
  it('should render without crashing', () => {
    const book = { Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 };
    const tree = renderer.create(<BookContent book={book} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
