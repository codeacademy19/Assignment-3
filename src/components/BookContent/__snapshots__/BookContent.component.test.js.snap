// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllBooksOfAnAuthor should render without crashing 1`] = `
<div
  className="mainContent"
>
  <div
    className="backgroundImage"
    style={
      Object {
        "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
      }
    }
  />
  <div
    className="title"
  >
    <span>
      HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
    </span>
  </div>
</div>
`;
